packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -376,6 +388,77 @@ contract L2Genesis is Script, Artifacts {
         vm.etch(Preinstalls.ENTRY_POINT, Preinstalls.ENTRY_POINT_DEPLOYED_BYTECODE);
     }
 
+    //////////////////////////////////////////////////////
+    /// Other
+    //////////////////////////////////////////////////////
+    function _getHardforkTime(uint256 hardforkTimeOffset, uint256 genesisTime) internal pure returns(uint256) {
+        uint256 _default = 0;
+        if (hardforkTimeOffset > 0) {
+            return genesisTime + hardforkTimeOffset;
+        }
+        return _default;
+    }
+
+    function _getChainConfig() internal returns (string memory) {
+        // TODO Go code is using block.Time() from provided *types.Block.
+        // devnetL1-template.json specifies "l2BlockTime": 2,
+        // so should genesisTime = cfg.l2BlockTime?
+        // block.timestamp = 1
+        uint256 genesisTime = block.timestamp;
+
+        string memory serializedConfigName = "chainConfig";
+        string memory json = "";
+        json = stdJson.serialize(serializedConfigName, "chainId", cfg.l2ChainID());
+        json = stdJson.serialize(serializedConfigName, "homesteadBlock", uint256(0));
+        // TODO DAOForkSupport is probably not supposed to be `false`, if `false` in
+        // deploy config, should we just not include it?
+        json = stdJson.serialize(serializedConfigName, "DAOForkSupport", false);
+        json = stdJson.serialize(serializedConfigName, "eip150Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "eip155Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "eip158Block", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "byzantiumBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "constantinopleBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "petersburgBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "istanbulBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "muirGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "berlinBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "londonBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "arrowGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "grayGlacierBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "mergeNetsplitBlock", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "terminalTotalDifficulty", uint256(0));
+        json = stdJson.serialize(serializedConfigName, "terminalTotalDifficultyPassed", true);
+        // TODO This was previously set as uint64, is it okay to be uint256?
+        json = stdJson.serialize(serializedConfigName, "bedrockBlock", cfg.l2GenesisBlockNumber());
+        json = stdJson.serialize(serializedConfigName, "regolithTime", _getHardforkTime(cfg.l2GenesisRegolithTimeOffset(), genesisTime));
```

Comment:
We will need to pull regolith time and afterwards from the deploy config. There are some go functions that are used. I believe that its based on an offset in the deploy config go code

Reply:
You have the `_getHardforkTime` which is good for this. 0 means that the fork is active at genesis. We will need to apply transactions to the state before outputting the state if ecotone is enabled at genesis. There is go code that has the set of ecotone txs and the spec also contains information on what they exactly are. I would prefer to have vm.sendRawTransaction` but that pull request was never fully implemented. It would allow us to just take the exact consensus rlp encoded tx and apply it directly to state. Since we do not have that, we will need to make calls. The calls must follow the go code/spec exactly